### ISSUE TITLE
Share video-links as Invidious urls

### ISSUE THREAD

**Issue by pred2k** [TeamNewPipe/NewPipe#2351]
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=74462002)](https://www.bountysource.com/issues/74462002-share-video-links-as-invidio-us-urls?utm_source=74462002&utm_medium=shield&utm_campaign=ISSUE_BADGE)

I imagine a setting to generate `invidio.us/watch?v=...` urls instead of `youtube.com/watch?v=...` when using the share function in NewPipe.

Was also already mentioned [here](https://github.com/TeamNewPipe/NewPipe/issues/1867#issuecomment-481164439), but that Issue is about implementing video playing via Invidious.

As you know opening links will already be supported (#1900) with v0.16.2.

**Comment by avoidr** [comment]
Also use invidio.us for "open in browser" function!

**Comment by PopeRigby** [comment]
This is a good idea, but maybe have it as a user setting. People might get annoyed if they don't want to share an invidio.us link.

**Comment by pred2k** [comment]
> This is a good idea, but maybe have it as a user setting. People might get annoyed if they don't want to share an invidio.us link.

Of course as a optional user setting.

And as extra maybe the possibility to choose between invidious instanes: https://github.com/omarroth/invidious/wiki/Invidious-Instances

**Comment by PopeRigby** [comment]
> And as extra maybe the possibility to choose between invidious instanes

That's a good idea. You could have a field where you can type in your instance, which would be nice for #1163.

**Comment by avoidr** [comment]
I also vote for invidio.us to be the default, with the option to pick youtube. No need to be so modest. invidious is superior technically and in terms of privacy and freedom.

**Comment by PB020** [comment]
Is this issue/bounty still open?

**Comment by Perflyst** [comment]
Yes

**Comment by src-r-r** [comment]
Probably best to shut it down, since...
![image](https://user-images.githubusercontent.com/1633901/97822663-8cf9cb00-1cae-11eb-8c06-83a965d354c6.png)

**Comment by pred2k** [comment]
> Probably best to close it, since...
> ![image](https://user-images.githubusercontent.com/1633901/97822663-8cf9cb00-1cae-11eb-8c06-83a965d354c6.png)
> Unless there's an alternative alternative?

invidio.us has shutdown, but there are other alternative working! instances. I would like leave the issue open and keep the bounty, just to promote these alternatives.

In the meantime i use the App UntrackMe for converting youtube (and twitter and instagram) urls into one of these privacy wrapping services.
https://f-droid.org/en/packages/app.fedilab.nitterizeme/
https://framagit.org/tom79/nitterizeme

**Comment by SameenAhnaf** [comment]
Kudos! Invidious has come back.

https://invidious.fdn.fr/

![IMG_20210501_102412](https://user-images.githubusercontent.com/79245641/116770718-7d91e980-aa67-11eb-98e1-7ca42918e007.jpg)

**Comment by TiA4f8R** [comment]
Now shared links should use the redirect.invidious.io domain.

**Comment by opusforlife2** [comment]
I think the user should be allowed to configure which domain to use. For example, when I'm sharing a link with my contacts, I would use the kavin.rocks domain directly, as it is in India.

The redirect URL would be far more useful if it automatically redirected a user to the appropriate instance based on their IP address.

**Comment by ZenithDev1** [comment]
> I put a $15 bounty on this https://www.bountysource.com/issues/74462002-share-video-links-as-invidio-us-urls

given link isn't working.

**Comment by pred2k** [comment]
> > I put a $15 bounty on this https://www.bountysource.com/issues/74462002-share-video-links-as-invidio-us-urls
> 
> given link isn't working.

Yeah, the bountysource was purchased by cryptodudes and is gone since a while.
The bounty is lost.

> As of June 2023, Bountysource appears to have stopped paying bounties to developers with verified claims. The Blockchain Group also appears to have stopped responding to Bountysource users.[[11]](https://en.m.wikipedia.org/wiki/Bountysource#cite_note-11)[[12]](https://en.m.wikipedia.org/wiki/Bountysource#cite_note-12)
> 
> In November 2023, the Bountysource parent company announced it had filed for bankruptcy.[[13]](https://en.m.wikipedia.org/wiki/Bountysource#cite_note-13)
> 
> As of May 2024, Bountysource website says "The site is temporarily down".[[14]](https://en.m.wikipedia.org/wiki/Bountysource#cite_note-14)

Source: https://en.m.wikipedia.org/wiki/Bountysource

I encourage everyone to support the transition to https://peertube.tv
The monopoly of YouTube has to collapse. NewPipe and alternative YouTube Frontends are awesome, but in the end they still feed the Big Tech empire.
